client/signup/config/steps-pure.js

```diff
@@ -111,7 +111,7 @@ export function generateSteps( {
 				headerText: i18n.translate( 'Getting ready to launch, pick a domain' ),
 				subHeaderText: i18n.translate( 'Select a domain name for your website' ),
 			},
-			dependencies: [ 'siteSlug' ],
+			dependencies: [ 'siteSlug', 'shouldHideFreePlan' ],
```

Comment:
This is incorrect - `shouldHideFreePlan` should be in `providesDependencies` and not `dependencies`. So, launch flow will not work.